src/main/java/org/apache/sling/jcr/repoinit/impl/NodeVisitor.java

```diff
@@ -57,18 +59,33 @@ public void visitCreatePath(CreatePath cp) {
             String parentPath = parentPathBuilder.toString();
             final String fullPath = String.format("%s/%s", parentPath, psd.getSegment());
             try {
-                if (session.itemExists(fullPath)) {
-                    log.info("Path already exists, nothing to do (and not checking its primary type for now): {}", fullPath);
-                } else {
+                final Node node;
+                if (session.nodeExists(fullPath)) {
+                    log.info("Node at {} already exists, checking/adjusting its types", fullPath);
+                    node = session.getNode(fullPath);
+                    if (psd.getPrimaryType() != null && !node.getPrimaryNodeType().getName().equals(psd.getPrimaryType())) {
+                        log.info("Adjusting primary type of node {} to {}", fullPath, psd.getPrimaryType());
+                        node.setPrimaryType(psd.getPrimaryType());
+                    }
+                } else if (!session.propertyExists(fullPath)) {
                     final Node parent = parentPath.equals("") ? session.getRootNode() : session.getNode(parentPath);
                     log.info("Creating node {} with primary type {}", fullPath, psd.getPrimaryType());
-                    Node node = addChildNode(parent, psd);
-                    List<String> mixins = psd.getMixins();
-                    if (mixins != null) {
-                        log.info("Adding mixins {} to node {}", mixins, fullPath);
-                        for (String mixin : mixins) {
-                            node.addMixin(mixin);
-                        }
+                    node = addChildNode(parent, psd);
+                    
+                } else {
+                    throw new RepoInitException("There is a property with the name of the to be created node already at " + fullPath + ", therefore bailing out here as potentially not supported by the underlying JCR");
```

Comment:
i don't think repo-init should check this. JCR specification leaves it to the implementation if same-name property/node are allowed and IMHO repo-init should delegate that verification.
i might be mistaken but i vaguely remember that oak actually supports it.

Reply:
you stated the opposite in https://issues.apache.org/jira/browse/SLING-10463.

Reply:
yes, that can happen.... depending on whether i am looking at a problem from a oak point of view or from a dev caught up in an AEM escalation.

Reply:
that makes it hard to discuss and leads to frustration ;-)